sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -506,6 +503,8 @@ export class EventHubSender extends LinkEntity {
           );
 
           try {
+            this._sender!.sendTimeoutInSeconds =
```

Comment:
Wait...doesn't this set the sendTimeoutInSeconds on the awaitable sender instance? Looking at the rhea-promise code it looks like this works now because there isn't any non-blocking work that occurs between when we call send and when rhea-promise calls setTimeout with this value, but there's no guarantee this won't change in the future.

Have we considered making a PR to add a timeout option to the awaitableSender send method? That would be safer since it'd be part of the public API. Otherwise I'm concerned that this could impact users that call sendBatch in parallel if rhea-promise ever changes their send implementation.

Reply:
That would indeed be safer. Am open to making a PR to rhea-promise for this

Reply:
https://github.com/amqp/rhea-promise/pull/53